pkg/codegen/pcl/binder_resource.go

```diff
@@ -430,6 +430,9 @@ func (b *binder) bindResourceBody(node *Resource) hcl.Diagnostics {
 				case "pluginDownloadURL":
 					t = model.StringType
 					resourceOptions.PluginDownloadURL = item.Value
+				case "deletedWith":
+					t = model.StringType
```

Comment:
Similar to my previous comment, should this be `model.DynamicType` to match what the `"parent"` case, which also takes a `Resource`?

```suggestion
					t = model.DynamicType
```